### DISCUSSION TITLE
What is the "Enable HTTP Referer header" toggle for?

### DISCUSSION THREAD

**Discussion by opusforlife2** [bromite/bromite#2406] [Q&A]
This toggle was on by default when I happened upon it.

I couldn't find any info on it. What is it for?

**Comment by Disqu3-mirOir-qt** [comment]
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referer

**Reply by opusforlife2** [reply]
I know what a referer is. What I'm asking is what exactly is the function of the toggle. For example, does it completely block all referers, which could break functionality,, or strip them of sensitive info, like a pref available in Firefox does?

**Reply by uazo** [reply]
@opusforlife2 follow https://github.com/bromite/bromite/search?q=HTTP+Referer&type=commits

```
The patch removes the referrals if the navigation is cross origin and occurs in the top frame.
We do not remove the value between the iframes because the referrals are statically defined
by the html and the javascript of the page, and therefore they do not say anything about the
user. Also, some services may not work, such as video iframes.
We also added a flag that completely removes referrals management, for users who know
what they are doing.
```

**Reply by opusforlife2** [reply]
@uazo Thank you for implementing it! Does the patch work when you toggle the setting on or off?

Also, could this be added to the wiki, @csagan5?

**Reply by ttntm** [reply]
Would love to know too. Maybe the UI label could be made a little more precise too?
